packages/venia-concept/src/actions/checkout/asyncActions.js

```diff
@@ -134,14 +158,21 @@ export const submitShippingMethod = payload =>
         }
 
         let billing_address = await retrieveBillingAddress();
-        const desiredShippingMethod = payload.formValues.shippingMethod;
+        const paymentMethod = await retrievePaymentMethod();
         const shipping_address = await retrieveShippingAddress();
+        const shipping_method = await retrieveShippingMethod();
 
-        // Remove extranneous properties from the billing address because the
-        // Magento backend will reject them.
-        billing_address = removeInvalidKeysFromAddress(billing_address);
+        if (billing_address.sameAsShippingAddress) {
+            billing_address = shipping_address;
+        } else {
+            billing_address = {
+                ...shipping_address,
+                ...billing_address
+            };
```

Comment:
This seems like it could have unintended effects. Is it possible for `shipping_address` to have property values that `billing_address` does not?

```es6
const a = { first: "Austin", middle: "Danger", last: "Powers" }
const b = { first: "Vanessa", last: "Kensington" }
const c = { ...a, ...b }

log(c) // { first: "Vanessa", middle: "Danger", last: "Kensington" }
```

Reply:
Yes, `shipping_address` definitely does have properties that `billing_address` does not (`firstname`, `lastname`, and `email`, specifically). I'm basically piggybacking those values from `shipping_address` here because the M2 will reject the `billing_address` if it doesn't contain them.

Reply:
I think it'd be fine to just explicitly copy those props into the object here rather than spreading the entire shipping address in that case.

Reply:
Resolved in https://github.com/magento-research/pwa-studio/pull/799/commits/6b34063e9a9b2d4e48eacd4c544f6dc1770e9d9e.